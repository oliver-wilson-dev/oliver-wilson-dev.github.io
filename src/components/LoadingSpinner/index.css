@import "../../styles/darkTheme.css";
@import "../../styles/lightTheme.css";

@keyframes spin {
  0% {
    transform: rotate(0deg);
  }
  
  100% {
    transform: rotate(360deg);
  }
}

.wrapper {
  margin: 10px 30px 10px 30px;
}

.loadingSpinner,
.loadingSpinner:after {
  border-radius: 50%;
  width: 10em;
  height: 10em;
  transition: opacity 1s;
}

.loadingSpinner {
  margin: 60px auto;
  font-size: 15px;
  position: relative;
  border-top: 10px solid var(--spinnerTrailColour);
  border-right: 10px solid var(--spinnerTrailColour);
  border-bottom: 10px solid var(--spinnerTrailColour);
  border-left: 10px solid var(--primaryBorderColour);
  transform: translateZ(0);
  animation: spin 1.5s infinite linear;
  
  &.hide {
    opacity: 0;
    &:after {
      display: none;
    }
  }
}

@media (min-width: 786px) {
  .wrapper {
    width: 826px;
  }
}