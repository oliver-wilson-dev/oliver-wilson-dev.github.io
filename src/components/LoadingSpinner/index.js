import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import styles from './index.css';


class LoadingSpinner extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      transitionDone: false
    };
  }

  onTransitionEnd() {
    this.setState({ transitionDone: true });
  }

  render() {
    const { props: { loading }, state: { transitionDone } } = this;

    return transitionDone
      ? null : (
        <div className={styles.wrapper}>
          <div
            className={classnames(styles.loadingSpinner, {
              [styles.hide]: !loading
            })}
            onTransitionEnd={() => this.onTransitionEnd()}
          />
        </div>
      );
  }
}

LoadingSpinner.propTypes = {
  loading: PropTypes.bool.isRequired
};

export default LoadingSpinner;
