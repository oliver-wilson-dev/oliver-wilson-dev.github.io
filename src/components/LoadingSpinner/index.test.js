import React from 'react';
import { shallow } from 'enzyme';
import LoadingSpinner from '.';

const render = () => shallow(<LoadingSpinner />);

describe('<LoadingSpinner />', () => {
  it('should render correctly', () => {
    expect(render().exists()).toBe(true);
    expect(render()).toMatchSnapshot();
  });
});
