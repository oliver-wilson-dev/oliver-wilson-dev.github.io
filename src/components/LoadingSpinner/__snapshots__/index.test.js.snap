// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingSpinner /> should render correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="loadingSpinner"
    onTransitionEnd={[Function]}
  />
</div>
`;

exports[`<LoadingSpinner /> when the component is not loading should render correctly 1`] = `""`;
